tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -99,24 +109,39 @@ def test_upgrade_downgrade_brokers(self, from_version, to_version):
         self.zk.start()
 
         # number of nodes needs to be >= 3 for the smoke test
-        self.kafka = KafkaService(self.test_context, num_nodes=3,
+        self.kafka = KafkaService(self.test_context, num_nodes=self.num_kafka_nodes,
                                   zk=self.zk, version=KafkaVersion(from_version), topics=self.topics)
         self.kafka.start()
 
         # allow some time for topics to be created
-        time.sleep(10)
+        wait_until(lambda: self.get_topics_count() >= (len(self.topics) * self.num_kafka_nodes),
+                   timeout_sec=60,
+                   err_msg="Broker did not create all topics in 60 seconds ")
 
         self.driver = StreamsSmokeTestDriverService(self.test_context, self.kafka)
+        self.driver.disable_auto_terminate()
+
         self.processor1 = StreamsSmokeTestJobRunnerService(self.test_context, self.kafka)
```

Comment:
nit: rename to `processor` because this test uses only one processor (the numbering is confusing otherwise)